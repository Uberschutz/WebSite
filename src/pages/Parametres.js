import React, { Component } from 'react';
import '../styles/bootstrap.css';
import  '../styles/Parameters.css';

import { Modal, Button } from 'reactstrap'
import Card from "reactstrap/es/Card";
import CardText from "reactstrap/es/CardText";
import CardTitle from "reactstrap/es/CardTitle";
import CardFooter from "reactstrap/es/CardFooter";

class Parametres extends Component {
	constructor(props) {
		super(props);
		this.state = {
			childrens: [],
			options: [],
			showModal: false,
			actual: {},
			name: '',
			age: '',
			id: null,
			state: 'Create'
		};
		this.toggleModal = this.toggleModal.bind(this);
	}

	componentWillMount() {
		const newChild = [{name: 'Thomas', age: '23'}, {name: 'Marianne', age: '21'}];

		this.setState({childrens: newChild});
		console.log('Here call to the user account API to load children'); // or in componentDidMount
	}

	// componentDidMount() {
	// 	const newChild = [{name: 'Thomas', age: '23'}];
	//
	// 	this.setState({childrens: newChild})
	// 	console.log('Here call to the user account API to load children'); // or in componentDidMount
	// }

	toggleModal() {
		this.setState({showModal: !this.state.showModal});
	}

	createChildren() {
		if (this.state.state === 'Create') {
			console.log('Here call to the user account API', this.state.name, this.state.age);
			const newChild = {name: this.state.name, age: this.state.age};
			let list = this.state.childrens;
			list.push(newChild);
			this.setState({childrens: list});
			this.toggleModal();
		} else {
			console.log('here', this.state.id);
			let childrens = this.state.childrens;
			childrens[this.state.id] = {name: this.state.name, age: this.state.age};
			this.setState({childrens: childrens});
			this.toggleModal();
		}
	}

	saveChildren() {
		console.log('Here call to the user account API');
	}

	editChildren(children) {
		console.log('Here call to the user account API to edit children', children);
		this.setState({name: children.name, age: children.age, state: 'Save', id: this.state.childrens.indexOf(children)});
		this.toggleModal();
	}

	deleteChildren(children) {
		console.log('Here call to the user account API to delete children', children);
		let list = this.state.childrens;
		const idx = list.indexOf(children);
		list.splice(idx, 1);
		this.setState({childrens: list});
	}

	setFirstName(name) {
		console.log('Here set children name', name);
		this.setState({name: name})
	}

	setAge(age) {
		console.log('Here set children age', age);
		this.setState({age: age});
	}

	render() {
        return (
            <div className="card align-card">
                {/*<button onClick={this.toggleModal}>Ajouter un enfant</button>*/}
	            <Modal isOpen={this.state.showModal} toggle={this.toggleModal} onClosed={() => this.setState({state: 'Created'})}>
					<button className="btn btn-danger" onClick={this.toggleModal}>Fermer</button>
					{/*<form onSubmit={() => this.createChildren()}>*/}
					<form> <br/>
					<div className="align-card">
						<label className="child-field">
							Name<br/>
							<input className="child-field" type="text" value={this.state.name} onChange={(name) => this.setFirstName(name.target.value)}/>
						</label>
						<br/>
						<label className="child-field">
							Age<br/>
							<input className="child-field" type="text" value={this.state.age} onChange={(age) => this.setAge(age.target.value)}/>
						</label>
					</div>
						<br/><br/>
						<div className="align-card">
							<Button className="save-child btn" onClick={() => this.createChildren()}>{this.state.state}</Button>
						</div>
						{/*<input type="submit" value="Submit" />*/}
					</form>
	            </Modal>
				{
					this.state.childrens.map((child, index) => {
						// console.log(child);
						return (
							<Card body inverse key={index} className="child-card">
								<CardTitle>{child.name}</CardTitle>
								<CardText>{child.age}</CardText>
								<CardFooter>
									<Button className="btn-mod btn btn-danger" onClick={() => this.deleteChildren(child)}>Delete</Button>
									<Button className="btn-mod btn btn-success" onClick={() => this.editChildren(child)}>Edit</Button>
								</CardFooter>
							</Card>
						);
					})
				}
				<Button className="btn btn-primary add-child" onClick={this.toggleModal}>New</Button>
            </div>
        )
    }
}

export default Parametres;