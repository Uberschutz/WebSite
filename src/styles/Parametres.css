.uber-color2 {
    background-color: black;
    color: white!important;
}