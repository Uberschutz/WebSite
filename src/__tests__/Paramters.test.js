import React from 'react';
import ReactDOM from 'react-dom';
import Parametres from '../pages/Parametres';

it('renders Parameters without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Parametres lang='fr'/>, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
});
