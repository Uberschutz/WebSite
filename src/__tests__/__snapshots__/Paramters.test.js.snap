// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Parameters without crashing 1`] = `
<div>
  <div
    class="card align-card"
  >
    <div
      class="child-card card text-white card-body"
    >
      <div
        class="card-title"
      >
        Prénom
         : 
        Thomas
      </div>
      <p
        class="card-text"
      >
        Age : 
        23
      </p>
      <div
        class="card-footer"
      >
        <button
          class="btn-mod btn btn-danger btn btn-secondary"
          type="button"
        >
          Supprimer
           
          <i
            aria-label="icon: delete"
            class="anticon anticon-delete size-icon"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="delete"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
              />
            </svg>
          </i>
        </button>
        <button
          class="btn-mod btn btn-success btn btn-secondary"
          type="button"
        >
          Éditer
           
          <i
            aria-label="icon: edit"
            class="anticon anticon-edit size-icon"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="edit"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
              />
            </svg>
          </i>
        </button>
      </div>
    </div>
    <div
      class="child-card card text-white card-body"
    >
      <div
        class="card-title"
      >
        Prénom
         : 
        Marianne
      </div>
      <p
        class="card-text"
      >
        Age : 
        21
      </p>
      <div
        class="card-footer"
      >
        <button
          class="btn-mod btn btn-danger btn btn-secondary"
          type="button"
        >
          Supprimer
           
          <i
            aria-label="icon: delete"
            class="anticon anticon-delete size-icon"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="delete"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
              />
            </svg>
          </i>
        </button>
        <button
          class="btn-mod btn btn-success btn btn-secondary"
          type="button"
        >
          Éditer
           
          <i
            aria-label="icon: edit"
            class="anticon anticon-edit size-icon"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="edit"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 0 0 0-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 0 0 9.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
              />
            </svg>
          </i>
        </button>
      </div>
    </div>
    <button
      class="btn btn-primary add-child btn btn-secondary"
      type="button"
    >
      Nouveau
       
      <i
        aria-label="icon: user-add"
        class="anticon anticon-user-add size-icon"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="user-add"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M678.3 642.4c24.2-13 51.9-20.4 81.4-20.4h.1c3 0 4.4-3.6 2.2-5.6a371.67 371.67 0 0 0-103.7-65.8c-.4-.2-.8-.3-1.2-.5C719.2 505 759.6 431.7 759.6 349c0-137-110.8-248-247.5-248S264.7 212 264.7 349c0 82.7 40.4 156 102.6 201.1-.4.2-.8.3-1.2.5-44.7 18.9-84.8 46-119.3 80.6a373.42 373.42 0 0 0-80.4 119.5A373.6 373.6 0 0 0 137 888.8a8 8 0 0 0 8 8.2h59.9c4.3 0 7.9-3.5 8-7.8 2-77.2 32.9-149.5 87.6-204.3C357 628.2 432.2 597 512.2 597c56.7 0 111.1 15.7 158 45.1a8.1 8.1 0 0 0 8.1.3zM512.2 521c-45.8 0-88.9-17.9-121.4-50.4A171.2 171.2 0 0 1 340.5 349c0-45.9 17.9-89.1 50.3-121.6S466.3 177 512.2 177s88.9 17.9 121.4 50.4A171.2 171.2 0 0 1 683.9 349c0 45.9-17.9 89.1-50.3 121.6C601.1 503.1 558 521 512.2 521zM880 759h-84v-84c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v84h-84c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h84v84c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-84h84c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </i>
    </button>
  </div>
</div>
`;
